src/main/java/entity/PassangerImpl.java

```diff
@@ -0,0 +1,107 @@
+
+package entity;
+
+    public class PassangerImpl implements ICar{
```

Comment:
If you want to use one parent for your entities it's better to make it abstract class.